x-pack/plugins/gis/public/shared/layers/layer.js

```diff
@@ -136,21 +139,24 @@ export class ALayer {
     return this._source.renderSourceSettingsEditor({ onChange });
   }
 
+  _findDataRequestForSource(sourceDataId) {
+    return this._dataRequests.find(dataRequest => dataRequest.getDataId() === sourceDataId);
+  }
+
+  getSourceDataRequest() {
+    return this._dataRequests.find(dataRequest => dataRequest.getDataId() === 'source');
+  }
+
   isLayerLoading() {
     return this._dataRequests.some(dataRequest => dataRequest.isLoading());
   }
 
   dataHasLoadError() {
-    return this._dataRequests.some(dataRequest => dataRequest.hasLoadError());
+    return this._descriptor && this._descriptor.errorState || false;
   }
 
   getDataLoadError() {
-    const loadErrors =  this._dataRequests
-      .filter(dataRequest => dataRequest.hasLoadError())
-      .map(dataRequest => {
-        return dataRequest._descriptor.dataLoadError;
-      });
-    return loadErrors.join(',');
+    return this.dataHasLoadError() ? this._descriptor.errorMessage : '';
```

Comment:
Maybe `errorMessage` should be an array `errors` so multiple message could be displayed. For example lets say you had a EMS vector layer with a join. Then EMS is not available and the index pattern of the join got deleted. Should have two messages - 1) Can not load EMS vector layer. 2) Can not access index pattern X

Reply:
I'm going to hold off on this since we're primarily focused on TMS for this issue, which _should_ have limited errors. I do think our messaging deserves further fleshing out. If we don't already have one, we likely need a future issue around better error messaging to the user. We have fairly robust messaging for the legacy maps now (some of it pending merge). We could port that over and rethink how errors are relayed in the process.